cmd/harmony/config.go

```diff
@@ -288,28 +357,62 @@ func registerDumpConfigFlags() error {
 	return cli.RegisterFlags(dumpConfigCmd, []cli.Flag{networkTypeFlag})
 }
 
-func loadHarmonyConfig(file string) (harmonyConfig, error) {
+func promptConfigUpdate() bool {
+	var readStr string
+	fmt.Println("Do you want to update config to the latest version: (Y/n)")
+	timeoutTimer := time.NewTimer(time.Second * 15)
+	read := make(chan string)
+	go func() {
+		fmt.Scanf("%s", &readStr)
+		read <- readStr
+	}()
+	select {
+	case <-timeoutTimer.C:
+		fmt.Println("Timed out - update manually with ./harmony config update [config_file]")
+		return false
+	case <-read:
+		readStr = strings.TrimSpace(readStr)
+		if len(readStr) > 1 {
+			readStr = readStr[0:1]
+		}
+		readStr = strings.ToLower(readStr)
+		return readStr == "y" || readStr == ""
+	}
+}
+
+func loadHarmonyConfig(file string) (harmonyConfig, string, error) {
 	b, err := ioutil.ReadFile(file)
 	if err != nil {
-		return harmonyConfig{}, err
+		return harmonyConfig{}, "", err
 	}
-
-	var config harmonyConfig
-	if err := toml.Unmarshal(b, &config); err != nil {
-		return harmonyConfig{}, err
+	config, migratedVer, err := migrateConf(b)
+	if err != nil {
+		return harmonyConfig{}, "", err
 	}
 
-	// Correct for old config version load (port 0 is invalid anyways)
-	if config.HTTP.RosettaPort == 0 {
-		config.HTTP.RosettaPort = defaultConfig.HTTP.RosettaPort
+	correctLegacyHarmonyConfig(&config)
```

Comment:
Done

Reply:
Would you please move this function to be only used when loading version <=v1.0.4? These are nasty "hack" solutions, and we don't want this value correction for future updates. Really appreciated.

Reply:
I have left correctLegacyHarmonyConfig intact since I wasn't sure it is not used to cover other cases (like user deleting part of the config), but In that case I  moved all the logic in 1.0.4 migration and remove correctLegacyHarmonyConfig altogether

Reply:
I gotta. Actually the content in `correctLegacyHarmonyConfig` is only used for v1.0.4. And `correctHarmonyConfig` shall be applied to all versions, which we shall add this function later.

You also provided a very good insight: what if a user deleted part of the config which is essential for starting the node? Shall we give it a default value and warn user of the absence of config data? Maybe I can create another bounty for this one later :)

Reply:
Yes it does have some use cases. User would be able to used very short config files. Basically your config file could only contain a few parameters that you want to override, everything else uses default values